aura.grammar

```diff
@@ -0,0 +1,56 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl ',' decl_list | epsilon
+
+type: i[1-9][0-9]* | f16 | f32 | f64 | f128 | '[' type ']' | '{' type ':' type '}'
+
+
+array : '[' identifier_list ']'
+
+assignment_statement: identifier type '=' expression
+                    | '!' assignment_statement
```

Comment:
The `!` operator in a declaration indicates the variable is mutable (Arua is an immutable-by-default language). It's similar to `mut` in rust.

As such, it's more of a property of the declaration. This is especially true since it is used within argument lists as well (e.g. `fn (a i32, !b i32, c i32)`, meaning only `b` can be re-assigned).

The assignment statement should really be something like

```
assignment_statement: (decl | identifier) assignment_operator expression
```

and `decl` should be modified to include `['!']`.